File: frontend-flutter/tripleu_gt/lib/services/auth_service.dart
import 'dart:convert';

import 'package:tripleu_gt/model/auth.dart';
import 'package:http/http.dart' as http;

class AuthService {
  String _url = '3.22.230.92:9100';
  String _path = '/oauth/token';

  Future<OAuth> login(String username, String password) async {

    String user = 'socialapp';
    String pass = '12345';
    String basicAuth =
      'Basic ' + base64Encode(utf8.encode('$user:$pass'));

  final url = Uri.http(_url, _path);
    final resp = await http.post(url, body: {
      "username": username,
      "password": password,
      "grant_type": "password"
    }, headers: <String, String>{
      "Authorization": basicAuth
    });
    final decoded = json.decode(resp.body);
        //rprint(decoded);

    final  oauth = new OAuth.fromMap(decoded);
    print(oauth);
    return (resp.statusCode == 200 ) ? oauth : null;
  }
}

File: frontend-angular-2/dashboard/src/app/services/auth.service.ts
import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { OAuth } from 'app/model/auth';
import { environment } from 'environments/environment';
import { map, catchError } from 'rxjs/operators';
import { BehaviorSubject, Observable, throwError } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class AuthService {

  private currentTokenSubject: BehaviorSubject<OAuth>;
  public  currentToken: Observable<OAuth>;

  constructor(private httpClient: HttpClient) {
    this.currentTokenSubject = new BehaviorSubject<OAuth>(JSON.parse(localStorage.getItem('currentUser')));
    this.currentToken = this.currentTokenSubject.asObservable();
  }

  public get currentUserValue(): OAuth {
    return this.currentTokenSubject.value;
}
  login(user: string, pass: string): Observable<OAuth> {
    const usr = "socialapp";
    const pss = "12345";
    const body = new HttpParams()
      .set("username", user)
      .set("password", pass)
      .set("grant_type", "password");
    return this.httpClient.post<OAuth>(`${environment.apiOAUTH}`, body, {
      headers: new HttpHeaders()
        .set("Authorization", `Basic ${btoa(`${usr}:${pss}`)}`)
    }).pipe(map(user => {
      localStorage.setItem('currentUser', JSON.stringify(user));


      this.currentTokenSubject.next(user);
      return user;
    }), catchError(this.handleError));
  }
  handleError(error) {
    let errorMessage = '';
    if (error.error instanceof ErrorEvent) {
      // client-side error
      errorMessage = `Error: ${error.error.message}`;
    } else {
      // server-side error
      errorMessage = `Error Code: ${error.status}\nMessage: ${error.message}`;
    }
    return throwError(errorMessage);
  }
  logout() {
    // remove user from local storage to log user out
    localStorage.removeItem('currentUser');
    this.currentTokenSubject.next(null);
}
}
